File: src/framework/examples/Singleton.java

public final class Singleton {
    private static Singleton singleton=null;
    private Singleton() {}
    public static Singleton get_instance() {
        if (singleton == null)
            singleton = new Singleton();
        return singleton;
    }

    public static void main(String[] args) {
        Singleton s=get_instance();
        Singleton s2=get_instance();
        System.out.println(s == s2);
    }
}

File: src/patterns/examples/singleton.cpp
#include <iostream>

using namespace std;

class Singleton {
    static Singleton* singleton;
    Singleton() {}
  public:
    static Singleton* get() {
        if (!Singleton::singleton)
            Singleton::singleton = new Singleton();
        return singleton;
    }
};

Singleton* Singleton::singleton;

int main() {
    Singleton* s{Singleton::get()};
    Singleton* s2{Singleton::get()};
    cout << (s == s2) << endl;
}
